Automattic/jetpack
# Search: improve Instant Search E2E tests stability

#### Changes proposed in this Pull Request:
- Added missing `await` for a function 
- Remove unnecessary `waitForPage` and `waitForNetworkIdle`
    - These two are the culprits making tests flaky, especially the latter, when there are requests that don't load (some AD requests are pretty suspicious), the function would waits till timeout and fails.
- Added `waitForInstantSearchReady` to check whether Instant Search is initialized okay to replace the checks above
- Removed obsolete code that operates search widgets, which now handled by [the auto config process](https://github.com/Automattic/jetpack/pull/22614).

#### Jetpack product discussion
p3QzjZ-1cj-p2
p1646416188445019-slack-jetpack-developers

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
- Follow [instruction](https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/tests/e2e/README.md) to setup local E2E env
- Run `pnpm test:run -- search.test.js`
- Ensure `4 passed`
- Ensure E2E tests on GitHub actions pass / restart several times, all search tests still pass
